schema/schema.hh

```diff
@@ -614,8 +614,24 @@ public:
     typedef std::ranges::subrange<iterator> iterator_range_type;
     typedef std::ranges::subrange<const_iterator> const_iterator_range_type;
 
-    static constexpr int32_t NAME_LENGTH = 48;
-
+    // The maximum allowed length for any schema name (table, keyspace, view, or index).
+    // This limit is primarily determined by the maximum table name length.
+    //
+    // When a new table is created, sstables::init_table_storage() generates a directory name
+    // by concatenating the full table name, a dash ('-'), and a 32-byte UUID. Since most Linux
+    // filesystems restrict a single filename component to 255 bytes, any table name longer than
+    // 222 bytes would result in a directory name exceeding this limit, causing mkdir() to fail.
+    //
+    // Additionally, we reserve 15 bytes for the "_scylla_cdc_log" suffix, which is appended to
+    // CDC-enabled table names (see cdc_log_suffix in cdc/log.cc). While this suffix
+    // only applies to table names, for simplicity, the same length restriction is enforced for
+    // keyspace, view, and index names as well.
+    // The reserved 15 bytes are for future use, allowing for potential future extensions without
+    // breaking existing schemas.
+    static constexpr size_t NAME_LENGTH = 255 - 32 // 32 bytes for UUID
+                                          - 1      // 1 byte for the dash (-) between the name and the UUID
+                                          - 15     // 15 bytes for the "_scylla_cdc_log" suffix.
+                                          - 15;    // 15 bytes reserved for future use.
```

Comment:
At this point I think it's worthwhile to just write the 255-32-1-15-15 explanation in a comment (as you did), but set the constant to the already calculated number 192 - instead of this calculation. We don't need future developers to do this calculation in their head just to figure out the number. Especially since we use the number "192" and not the calculation in documentation in tests.